Sources/NIO/EventLoopFuture+Combine.swift

```diff
@@ -0,0 +1,134 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Combine)
```

Comment:
This means that NIO will have different public API depending on whether Combine is available or not.

It feels to me like this code belongs in `swift-nio-extras` or a new `swift-nio-combine` project, so that NIO remains cross-platform.

Reply:
While that’s true, I don’t know how big of an issue that is. These types are not actually usable without `Combine`: even if you called this API on a platform without `Combine`, you wouldn’t be able to do anything with the returned type, and in order to actually compute on the value you’d have to `import Combine`, which would naturally fail.

I’m interested to see if anyone can actually write an example of this that is both a) reasonable, and b) does not compile on both platforms due to the differing NIO public API. My suspicion is that this can’t be done.

Regardless, I’m open to making a change here if the community and other maintainers think it’s necessary.

Reply:
Thanks. This is a SemVer minor change for Darwin, but no change at all for Linux, which seems odd. Hypothetically, would you release SwiftNIO 2.3 containing just this PR?

Reply:
As a SemVer minor. The appropriate semver status for a change is the max of any hypothetical change on any platform.

Reply:
> I'll be pulling this out to a new target in the near future.

Was this ever done? I'm erasing `EventLoopFuture`s to `AnyPublisher`s in a project of mine and could use something like this.

Reply:
Hey JP, no, this wasn't done. If you wanted to do the work you could port this change to a new `NIOCombineCompat` target.